Jenkinsfile

```diff
@@ -1,134 +1,198 @@
+  currentBuild.result = 'FAILURE'
+  error("${reason}")
 }
 
 node('lisk-explorer-01'){
-  lock(resource: "lisk-explorer-01", inversePrecedence: true) {
+  try {
     stage ('Prepare Workspace') {
-      cleanUp()
+      deleteDir()
       checkout scm
     }
 
     stage ('Build Dependencies') {
       try {
-        sh '''#!/bin/bash
+        sh '''
 
         # Install Deps
         npm install
+        ./node_modules/protractor/bin/webdriver-manager update
+        wget https://downloads.liskwallet.net/lisk/redis/redis-3.2.9-Linux-x86_64.tar.gz
+        tar -zvxf redis-3.2.9-Linux-x86_64.tar.gz
         '''
       } catch (err) {
-        currentBuild.result = 'FAILURE'
-        error('Stopping build, installation failed')
+        echo "Error: ${err}"
+        fail('Stopping build, installation failed')
       }
     }
 
     stage ('Run Eslint') {
       try {
-        sh '''#!/bin/bash
+        sh '''
         # Run Eslint
         npm run eslint
         '''
       } catch (err) {
-        currentBuild.result = 'FAILURE'
-        error('Stopping build, Eslint failed')
+        echo "Error: ${err}"
+        fail('Stopping build, webpack failed')
       }
     }
 
     stage ('Run Webpack Build') {
       try {
-        sh '''#!/bin/bash
+        sh '''
         # Build Bundles
         npm run build
         '''
       } catch (err) {
-        currentBuild.result = 'FAILURE'
-        error('Stopping build, webpack failed')
+        echo "Error: ${err}"
+        fail('Stopping build, webpack failed')
+      }
+    }
+
+    stage ('Start Redis') {
+      try {
+        sh '''
+        N=${EXECUTOR_NUMBER:-0}
+        ./redis-server --port 700$N > redis$N.log &
+        cp test/config.test ./config.js
+
+        '''
+      } catch (err) {
+        echo "Error: ${err}"
+        fail('Stopping build, webpack failed')
       }
     }
 
     stage ('Build Candles') {
       try {
-        sh '''#!/bin/bash
+        sh '''
+        N=${EXECUTOR_NUMBER:-0}
         # Generate market data
-        grunt candles:build
+        REDIS_PORT=700$N grunt candles:build
         '''
       } catch (err) {
-        currentBuild.result = 'FAILURE'
-        error('Stopping build, candles build failed')
+        echo "Error: ${err}"
+        fail('Stopping build, candles build failed')
       }
     }
 
     stage ('Start Lisk ') {
       try {
-        sh '''#!/bin/bash
-        cp test/config_lisk.json ~/lisk-test/config.json
-        bash ~/lisk-test/lisk.sh rebuild -f ~/lisk-test/blockchain_explorer.db.gz
+
+        sh '''
+        N=${EXECUTOR_NUMBER:-0}
+        # work around core bug: config.json gets overwritten; use backup
+        cp test/config_lisk.json ~/lisk-test/config_$N.json
+        cd ~/lisk-test
+        # change core port
+        sed -i -r -e "s/^(.*ort\\":) 4000,/\\1 400$N,/" config_$N.json
```

Comment:
you could probably use JQ instead of sed to do this.

Reply:
We should  also leave core permanently running and remove alot of these steps, it doesnt hurt that more than one node would query it for testing.